### PR TITLE
fix(config): detect missing RENOVATE_CONFIG_FILE

### LINKED ISSUE #7620: renovate silently ignores config if RENOVATE_CONFIG_FILE points to a non-existing file or config.js throws on evaluation
**What Renovate type, platform and version are you using?**

latest renovate version on self-hosted GitLab instance

**Describe the bug**

When running `renovate-config-validator` on the following `config.js` file (which throws on evaluation if CI_API_V4_URL is undefined):
```
module.exports = {
    platform: 'gitlab',
    ...
    endpoint: CI_API_V4_URL.replace(...)
    ...
};
```
it incorrectly prints
```
Validating config.js
OK
```
instead of failing and/or re-throwing the exception thrown by the (incorrect) config.js file.

The same happens if we set the `RENOVATE_CONFIG_FILE` env variable to a non-existing filename.
I would expect that `renovate-config-validator` also throws or fails in this case.

This is because of
https://github.com/renovatebot/renovate/blob/c2ebd71732e420c2bc8c80af526ccc7a644b1775/lib/config/file.ts#L18
only checking for errors of type `SyntaxError` and proceeding silently on all other types of errors (evaluation exceptions, fileNotFound exception etc.)

### PR DESCRIPTION
**Pull request by olegkrivtsov**
## Changes:

Renovate silently ignores config if RENOVATE_CONFIG_FILE points to a non-existing file or config.js throws on evaluation #7620

## Context:

Closes #7620

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/global/config/parse/file.ts [hunk 1]
```diff
@@ -45,6 +45,12 @@ export async function getConfig(env: NodeJS.ProcessEnv): Promise<AllConfig> {
     if (err instanceof SyntaxError || err instanceof TypeError) {
       logger.fatal(`Could not parse config file \n ${err.stack}`);
       process.exit(1);
+    } else if (err instanceof ReferenceError) {
```
Comment (viceice) [inline lib/workers/global/config/parse/file.ts:48]: I think we should check if the file exists earlier. Remember that the file is optional

Comment (olegkrivtsov) [reply lib/workers/global/config/parse/file.ts:55]: Hi @viceice I've reviewed the issue description another time and also tried to take your suggestion into account. So I came to this solution:

1. Consider the config file optional and don't complaint/exit if it doesn't present.
2. If `enc.RENOVATE_CONFIG_FILE` is specified, check if that file exists and exit with fatal error if it desn't exist.
3. When reading JS config file, check for `ReferenceError` and exit with fatal error if some varialbe can't be resolved.

File: lib/workers/global/config/parse/file.ts [hunk 2]
```diff
@@ -36,6 +37,12 @@ export async function getConfig(env: NodeJS.ProcessEnv): Promise<AllConfig> {
   if (!upath.isAbsolute(configFile)) {
     configFile = `${process.cwd()}/${configFile}`;
   }
+
+  if (env.RENOVATE_CONFIG_FILE && !fs.existsSync(configFile)) {
```
Comment (rarkins) [inline lib/workers/global/config/parse/file.ts:41]: Can you use a non-sync alternative?

File: lib/workers/global/config/parse/file.ts [hunk 3]
```diff
@@ -36,6 +37,12 @@ export async function getConfig(env: NodeJS.ProcessEnv): Promise<AllConfig> {
...
+    logger.fatal(`Custom config file ${configFile} must exist`);
```
Comment (rarkins) [inline lib/workers/global/config/parse/file.ts:42]:
```suggestion
    logger.fatal(`Custom config file ${configFile} specified in RENOVATE_CONFIG_FILE must exist`);
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/workers/global/config/parse/file.spec.ts [hunk 4]
```diff
@@ -92,6 +92,30 @@ describe('workers/global/config/parse/file', () => {
...
+      const configFile = upath.resolve(
+        __dirname,
+        './__fixtures__/config-ref-error.js-invalid'
+      );
+      const tmpDir = tmp.path;
+      if (!fs.existsSync(tmpDir)) {
+        fs.mkdirSync(tmpDir);
+      }
+      const tmpConfigFile = upath.resolve(tmpDir, 'config-ref-error.js');
+      fs.copyFileSync(configFile, tmpConfigFile);
```
Comment (viceice) [inline lib/workers/global/config/parse/file.spec.ts:109]: please use async functions

File: lib/workers/global/config/parse/file.ts (hunk 3 above)
Comment (viceice) [reply lib/workers/global/config/parse/file.ts:42]:
```suggestion
    logger.fatal({ configFile }, `Custom config file specified in RENOVATE_CONFIG_FILE must exist`);
```

Use logger metadata

**Review by viceice — APPROVED** [review]
@rarkins I think this is ready,

**Review by rarkins — APPROVED** [review]

**Pull request merged**